### PR TITLE
Javascript javascript3 week3

### PR DESCRIPTION
**Pull request by Mabonye**
Javascript3 week3 homework

### CODE REVIEW

File: javascript/javascript3/week3/main.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        this.signupButton.addEventListener("click", this.saveUserData.bind(this));
+        this.loginButton.addEventListener("click", this.login.bind(this));
+    }
+
+    /*   
+        Method for fetching data for users who signed up
+    */
+    async getUserData() {
+        const response = await fetch(this.saveDataAPI);
+        const result = await response.json();
```
Comment (Jane-API-SG) [inline javascript/javascript3/week3/main.js]: Both of these async calls can fail and end up returning an unhandled promise. They should be wrapped in a try/catch.

File: javascript/javascript3/week3/screenshot.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+            console.error(err);
+        }
+    }
+    
+    /*
+        Method for fetching saved screenshots
+    */
+    async getSavedScreenshot() {
+        let response = await fetch(this.saveScreenshotAPI);
+        let result = await response.json();
```
Comment (Jane-API-SG) [inline javascript/javascript3/week3/screenshot.js]: These should be wrapped in a try/catch as well.

**Review by Jane-API-SG — APPROVED** [review]
Hi Jemima,
It looks really good. The only thing I have commented on is a few missing try/catch statements. But I see you using it correctly throughout the rest of the code.

Really good work!
Sorry for getting back to you late.

Best Jane

**Comment by Mabonye** [comment]
> Hi Jemima, It looks really good. The only thing I have commented on is a few missing try/catch statements. But I see you using it correctly throughout the rest of the code.
> 
> Really good work! Sorry for getting back to you late.
> 
> Best Jane

Hello Jane,
Thanks for your time and comments, I'll go through them and modify my homework:)
Regards,
Jemima.

**Comment by Mabonye** [comment]
> Hi Jemima, It looks really good. The only thing I have commented on is a few missing try/catch statements. But I see you using it correctly throughout the rest of the code.
> 
> Really good work! Sorry for getting back to you late.
> 
> Best Jane

Hello Jane:)
Thank you for your time and comments, will go through the and modify my homework.
Regards,
Jemima.

**Pull request merged**